src/main/ruby/core/truffle/regexp_operations.rb

```diff
@@ -23,15 +23,14 @@ def self.match(re, str, pos=0)
     def self.last_match(a_binding)
       Truffle::KernelOperations.frame_local_variable_get(:$~, a_binding)
     end
+    Truffle::Graal.always_split(method(:last_match))
 
     def self.set_last_match(value, a_binding)
       unless value.nil? || Truffle::Type.object_kind_of?(value, MatchData)
         raise TypeError, "Wrong argument type #{value} (expected MatchData)"
       end
-      Truffle::KernelOperations.frame_local_variable_set(:$~, a_binding, value)
+      Truffle::KernelOperations.frame_local_variable_set(:$~, a_binding, value) if a_binding
```

Comment:
Why the `if` here? Is there a caller using a `nil` Binding?

Reply:
Yes. Procs created from methods don't have a declaration frame (this is a bug in our code). I tried changing that, but it didn't quite work as we have assumptions elsewhere about what a declaration frame contains. I'll do a PR later when I've managed to bottom out those issues properly.

Reply:
OK, please leave TODO here then to remember to cleanup.